include/vikunja/access/BaseStrategy.hpp

```diff
@@ -0,0 +1,89 @@
+/* Copyright 2022 Hauke Mewes, Jonas Schenke, Simeon Ehrig
+ *
+ * This file is part of vikunja.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/alpaka.hpp>
+
+namespace vikunja::MemAccess
+{
+    //! Base class to implement memory access strategy.
+    //!
+    //! \tparam TIdx Index type
+    template<typename TIdx>
+    class BaseStrategy
+    {
+    protected:
+        TIdx m_index;
+        TIdx const m_maximum;
+
+    public:
+        //-----------------------------------------------------------------------------
+        //! Constructor.
+        //!
+        //! \param index The index.
+        //! \param maximum The first index outside of the iterator memory.
+        ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE BaseStrategy(TIdx const index, TIdx const maximum)
+            : m_index(index)
+            , m_maximum(maximum)
+        {
+        }
+
+        ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE BaseStrategy(const BaseStrategy& other) = default;
+
+        ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE auto operator==(const BaseStrategy& other) const -> bool
```

Comment:
@bernhardmgruber recommends to make comparison operators a `friend`:

```suggestion
        ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE friend auto operator==(const BaseStrategy& other) const -> bool
```

Equivalent for the others.